engine/src/main/java/com/datatorrent/stram/webapp/StramWebServices.java

```diff
@@ -767,8 +900,17 @@ public JSONObject getPhysicalOperatorProperties(@PathParam("operatorId") int ope
   @Produces(MediaType.APPLICATION_JSON)
   public JSONObject getLogicalPlan() throws JSONException, IOException
   {
-    LogicalPlan lp = dagManager.getLogicalPlan();
-    return new JSONObject(objectMapper.writeValueAsString(LogicalPlanSerializer.convertToMap(lp)));
+    return new JSONObject(objectMapper.writeValueAsString(LogicalPlanSerializer.convertToMap(
+        dagManager.getLogicalPlan(), false)));
+  }
+
+  @GET
+  @Path(PATH_LOGICAL_PLAN + "/flatten")
+  @Produces(MediaType.APPLICATION_JSON)
+  public JSONObject getLogicalPlanFlatten() throws JSONException, IOException
+  {
+    return new JSONObject(objectMapper.writeValueAsString(LogicalPlanSerializer.convertToMap(
+        dagManager.getLogicalPlan(), true)));
   }
```

Comment:
again, flatten should not be part of the resource path.  is there support for getting individual operator/module that is arbitrarily deep under specific modules?